pkg/chains/storage/tekton/tekton.go

```diff
@@ -77,57 +80,81 @@ func (b *Backend) Type() string {
 }
 
 // retrieveAnnotationValue retrieve the value of an annotation and base64 decode it if needed.
-func (b *Backend) retrieveAnnotationValue(ctx context.Context, tr *v1beta1.TaskRun, annotationKey string, decode bool) (string, error) {
+func (b *Backend) retrieveAnnotationValue(ctx context.Context, clientSet versioned.Interface, obj objects.TektonObject, annotationKey string, decode bool) (string, error) {
 	// Retrieve the TaskRun.
-	b.logger.Infof("Retrieving annotation %q on TaskRun %s/%s", annotationKey, tr.Namespace, tr.Name)
-	tr, err := b.pipelienclientset.TektonV1beta1().TaskRuns(tr.Namespace).Get(ctx, tr.Name, v1.GetOptions{})
```

Comment:
It looks like `retrieveAnnotationValue` is only ever called by `RetrieveSignatures` and `RetrievePayloads` below. The first is not called anywhere within Chains. The second is called from `VerifyTaskRun` in `pkg/chains/verifier.go`. `VerifyTaskRun` is not called from anywhere.

This is either meant to be used by something outside of Chains, or it is dead code. I wonder if it was added to assist in the cli? https://github.com/tektoncd/cli/blob/main/pkg/chain/chain.go

Reply:
It looks like the `RetrieveSignatures` and `RetrievePayloads` are also used as helper functions for testing the Tekton backend. Maybe they were placed here in case they had value in the future? As of now they would probably fit better in a testing utilities directory.

Reply:
@wlynch @chuangw6 Is keeping `VerifyTaskRun` required if it isn't being used anywhere? Is it being imported by other projects? It would make the most sense to clean this up and separate `RetrieveSignatures` and `RetrievePayloads` into the internal testing directory.

Reply:
> @chuangw6 Do you remember why we're getting the latest annotations here?
> 
> I'm wondering why the annotations of the state we're reconciling isn't sufficient 🤔

I think it's only for retrieving the payload from taskrun's annotation.


> @wlynch @chuangw6 Is keeping VerifyTaskRun required if it isn't being used anywhere? Is it being imported by other projects? It would make the most sense to clean this up and separate RetrieveSignatures and RetrievePayloads into the internal testing directory.

It's not required nor used by other projects AFAIK.

Reply:
> @chuangw6 Do you remember why we're getting the latest annotations here?
> 
> I'm wondering why the annotations of the state we're reconciling isn't sufficient 🤔

I think it's only for retrieving the payload from taskrun's annotation.


> @wlynch @chuangw6 Is keeping VerifyTaskRun required if it isn't being used anywhere? Is it being imported by other projects? It would make the most sense to clean this up and separate RetrieveSignatures and RetrievePayloads into the internal testing directory.

It's not required nor used by other projects AFAIK.

Reply:
Actually what's interesting is `RetrieveSignatures` and `RetrievePayloads` has been implemented by every backend but none are used anywhere except tests

Reply:
Talked to @wlynch a while ago about the usage of these retrieval functions. There might be potential usage in future when people want to add CLI capability to retrieve & verify attestations. But other than that, I agree they are just used in tests for now.

Reply:
okay i left them in for now. Removing such a large amount of code that may have use in future seems out of scope for this PR